src/main/service/sessionManagementService.ts

```diff
@@ -1,63 +1,49 @@
-import moment from 'moment';
 import {allAdminRoles, checkRoles} from '../authentication/authenticationHandler';
 import {B2C_ADMIN_URL, B2C_URL, FRONTEND_URL} from '../helpers/envUrls';
 
 const authenticationConfig = require('../authentication/authentication-config.json');
-const defaultSessionExpiry = 60 * 60 * 1000; // default to 1 hour
 
 export class SessionManagementService {
-  public logOut(req, res, adminWrongFlow): void {
+  public logOut(req, res, adminWrongFlow, isSessionExpired = false): void {
     // For cookie-session, the request session needs to be destroyed by setting to null upon logout
     req.session = null;
     res.clearCookie('session');
+    res.redirect(this.logOutUrl(checkRoles(req, allAdminRoles), adminWrongFlow, isSessionExpired, req.lng));
+  }
+
+  private logOutUrl(isAdmin: boolean, adminWrongFlow: boolean, isSessionExpired: boolean, language: string): string {
     let b2cUrl;
     let b2cPolicy;
+
     if (adminWrongFlow) {
       b2cUrl = B2C_URL;
       b2cPolicy = authenticationConfig.POLICY;
     } else {
-      b2cUrl = checkRoles(req, allAdminRoles) ? B2C_ADMIN_URL : B2C_URL;
-      b2cPolicy = checkRoles(req, allAdminRoles) ? authenticationConfig.ADMIN_POLICY : authenticationConfig.POLICY;
+      b2cUrl = isAdmin ? B2C_ADMIN_URL : B2C_URL;
+      b2cPolicy = isAdmin ? authenticationConfig.ADMIN_POLICY : authenticationConfig.POLICY;
     }
 
-    const encodedSignOutRedirect = encodeURIComponent(this.getLogOutRedirectUrl(req, adminWrongFlow));
-    res.redirect(`${b2cUrl}/${b2cPolicy}/oauth2/v2.0/logout?post_logout_redirect_uri=${encodedSignOutRedirect}`);
+    const encodedSignOutRedirect = encodeURIComponent(this.logOutRedirectUrl(isAdmin, adminWrongFlow, isSessionExpired, language));
+    return `${b2cUrl}/${b2cPolicy}/oauth2/v2.0/logout?post_logout_redirect_uri=${encodedSignOutRedirect}`;
   }
 
-  public handleSessionExpiry(req, res): boolean {
-    if(this.isSessionExpired(req)) {
-      this.logOut(req, res, false);
-      return true;
-    }
-    return false;
-  }
+  private logOutRedirectUrl(isAdmin: boolean, adminWrongFlow: boolean, isSessionExpired: boolean, language: string): string {
+    const url = new URL(`${FRONTEND_URL}/${this.getRedirectionPath(adminWrongFlow, isSessionExpired)}`);
+    url.searchParams.append('lng', language);
 
-  private isSessionExpired(req): boolean {
```

Comment:
I'm wondering if we still want this in

If someone doesn't have JS enabled then the frontend triggers won't hit so I think we still need something to catch it if the session expires

Reply:
You have a good point here Chris. If someone has JS enabled this bit of code wont hit before the user would have already been auto signed out. So this bit of code is only going to matter for those without JS so the got signed out too if they navigate somewhere else after the session expired. Have added it back in :)